src/chart-addons/stacked-legend.js

```diff
@@ -98,7 +98,42 @@ function setColorScaleDomain_v1(domain) {
   }
 }
 
-export function getLegendStateFromChart(chart, useMap, selectedLayer) {
+async function getTopValues(layer, size) {
+  const OFFSET = 0
+  const dimension = layer?.getState()?.encoding?.color?.field
+  if (dimension) {
+    return layer
+      .crossfilter()
+      .dimension(dimension)
+      .group()
+      .topAsync(size, OFFSET)
+      .then(results => {
+        if (results) {
+          return results.map(result => result.key0)
+        } else {
+          return null
+        }
+      })
+      .catch(error => null)
+  } else {
+    return Promise.resolve(null)
```

Comment:
I didn't think you had to explicitly return a promise here, since its an async function. Maybe charting babel config is weird tho

```suggestion
    return null
```